src/test/java/com/microsoft/azure/proton/transport/proxy/impl/ProxyResponseImplTest.java

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.proton.transport.proxy.impl;
+
+import com.microsoft.azure.proton.transport.proxy.HttpStatusLine;
+import com.microsoft.azure.proton.transport.proxy.ProxyResponse;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.microsoft.azure.proton.transport.proxy.impl.StringUtils.NEW_LINE;
+
+public class ProxyResponseImplTest {
+    /**
+     * Verifies that it successfully parses a valid HTTP response.
+     */
+    @Test
+    public void validResponse() {
```

Comment:
Do you have any test cases where the message takes multiple frames and has that continuation?

Reply:
Sorry I forgot this most important test cases. already added.